src/Joonasw.AspNetCore.SecurityHeaders/ExpectCT/ExpectCTMiddleware.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+
+namespace Joonasw.AspNetCore.SecurityHeaders.ExpectCT
+{
+    public class ExpectCTMiddleware
+    {
+        private const string HeaderName = "Expect-CT";
+        private readonly RequestDelegate _next;
+        private readonly string _headerValue;
+
+        public ExpectCTMiddleware(RequestDelegate next, IOptions<ExpectCTOptions> options)
+        {
+            _next = next;
+            _headerValue = options.Value.HeaderValue;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            //Expect-CT can only be applied to secure requests according to spec
+            if (context.Request.IsHttps)
+            {
+                context.Response.Headers.Add(HeaderName, _headerValue);
```

Comment:
We need to check if the response already contains an Expect-CT header before trying to add it